src/engraving/dom/cmd.cpp

```diff
@@ -3315,6 +3316,48 @@ void Score::cmdIncDecDuration(int nSteps, bool stepDotted)
     nextInputPos(cr, false);
 }
 
+//---------------------------------------------------------
+//   extendToNextNote
+//---------------------------------------------------------
+
+void Score::cmdExtendToNextNote()
+{
+    Fraction startTick = selection().tickStart();
+    Fraction endTick = selection().tickEnd();
+    staff_idx_t startStaff = selection().staffStart();
+    staff_idx_t endStaff = selection().staffEnd();
+
+    for (EngravingItem* el : selection().elements()) {
+        if (el->isNote() || el->isChord()) {
+            Note* on = toNote(el);
+            ChordRest* ocr = toChordRest(on->chord());
+            ChordRest* nextCR = nextChordRest(ocr);
+
+            while (nextCR->type() == ElementType::REST) {
+                Rest* r = toRest(nextCR);
+                ocr = prevChordRest(toChordRest(r));
+                if (ocr->tuplet() || r->tuplet()) {
+                    addChord(r->tick(), r->ticks(), toChord(ocr), true, r->tuplet());
+                } else {
+                    ocr = prevChordRest(toChordRest(r));
+                    auto num = r->ticks().denominator() * ocr->ticks().numerator() + ocr->ticks().denominator() * r->ticks().numerator();
+                    auto den = ocr->ticks().denominator() * r->ticks().denominator();
+                    changeCRlen(ocr, Fraction(num, den));
```

Comment:
Given that `Fraction` has a `+` operator, isn't is possible to write
```suggestion
                    changeCRlen(ocr, ocr->ticks() + r->ticks());
```
?

Reply:
yes, it works! Done.